packages/app/src/cli/services/logs.test.ts

```diff
@@ -57,9 +57,55 @@ describe('logs', () => {
     // Then
     expect(spy).toHaveBeenCalled()
   })
+
+  test('should raise error when app has no valid sources', async () => {
+    // Given
+    await setupDevContext([])
+    const spy = vi.spyOn(renderLogs, 'renderLogs')
+
+    // When
+    await expect(() => {
+      return logs({
+        reset: false,
+        format: 'text',
+        apiKey: 'api-key',
+        directory: 'directory',
+        storeFqdn: 'store-fqdn',
+        sources: ['extensions.source'],
+        status: 'status',
+        configName: 'config-name',
+        userProvidedConfigName: 'user-provided-config-name',
+      })
+    }).rejects.toThrowError(
+      'This app has no log sources. Learn more about app logs at https://shopify.dev/docs/apps/build/functions/log-streaming-and-replay.',
```

Comment:
cc @nickwesselman this is raised if `app logs` is called in a scenario where we know that a log will never be returned. Thoughts on copy here and where to link the user to explain why logs won't return anything?

Reply:
Why don't we link them to the reference docs for the `app logs` command and describe the behavior there? The URL will be

https://shopify.dev/docs/api/shopify-cli/app/app-logs

Reply:
Sure, sounds good.